atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -0,0 +1,324 @@
+
+    def adjust_for_autorebuild(self):
+        """Ignore pre-filled signing_intent and compose_ids for autorebuids
+
+        Auto rebuilds are expected to use a know configuration. The parameters
+        signing_intent and compose_ids are meant for one-off build calls. This
+        method ensure that these parameters are ignored for autorebuilds.
+        """
+        if not is_rebuild(self.workflow):
+            return
+
+        if self.signing_intent:
+            self.log.info('Autorebuild detected: Ignoring signing_intent plugin parameter')
+            self.signing_intent = None
+
+        if self.compose_ids:
+            self.log.info('Autorebuild detected: Ignoring compose_ids plugin parameter')
+            self.compose_ids = tuple()
+
+    def read_configs(self):
+        self.odcs_config = get_config(self.workflow).get_odcs_config()
+        if not self.odcs_config:
+            raise AbortResolveComposesPlugin('ODCS config not found')
+
+        workdir = self.workflow.source.get_build_file_path()[1]
+        file_path = os.path.join(workdir, self.REPO_CONFIG)
+        data = None
+        if os.path.exists(file_path):
+            with open(file_path) as f:
+                data = (yaml.load(f) or {}).get('compose')
+
+        if not data and not self.compose_ids:
+            raise AbortResolveComposesPlugin('"compose" config not set and compose_ids not given')
+
+        self.compose_config = ComposeConfig(data, self.odcs_config)
+
+    def adjust_compose_config(self):
+        if self.signing_intent:
+            self.compose_config.set_signing_intent(self.signing_intent)
+
+        if self.koji_target:
+            target_info = self.koji_session.getBuildTarget(self.koji_target, strict=True)
+            self.compose_config.koji_tag = target_info['build_tag_name']
+
+        self.adjust_signing_intent_from_parent()
+
+    def adjust_signing_intent_from_parent(self):
+        plugin_result = self.workflow.prebuild_results.get(PLUGIN_KOJI_PARENT_KEY)
+        if not plugin_result:
+            self.log.debug("%s plugin didn't run, signing intent will not be adjusted",
+                           PLUGIN_KOJI_PARENT_KEY)
+            return
+
+        build_info = plugin_result['parent-image-koji-build']
+
+        try:
+            parent_signing_intent_name = build_info['extra']['image']['odcs']['signing_intent']
+        except (KeyError, TypeError):
+            self.log.debug('Parent koji build, %s(%s), does not definie signing_intent. '
+                           'Cannot adjust adjust for current build.',
+                           build_info['nvr'], build_info['id'])
+            return
+
+        self._parent_signing_intent = (self.odcs_config
+                                       .get_signing_intent_by_name(parent_signing_intent_name))
+
+        current_signing_intent = self.compose_config.signing_intent
+
+        new_signing_intent = min(self._parent_signing_intent, current_signing_intent,
```

Comment:
Maybe add a comment here explaining we're calculating the least restrictive signing intent?

Reply:
Done.